pandas/core/groupby/generic.py

```diff
@@ -1203,16 +1203,27 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
 
             values = [x if (x is not None) else backup for x in values]
 
-        v = values[0]
-
-        if not isinstance(v, (np.ndarray, Index, Series)) and self.as_index:
+        if isinstance(first_not_none, (np.ndarray, Index)):
+            # GH#1738: values is list of arrays of unequal lengths
+            #  fall through to the outer else clause
+            # TODO: sure this is right?  we used to do this
+            #  after raising AttributeError above
+            return self.obj._constructor_sliced(
+                values, index=key_index, name=self._selection_name
+            )
+        elif not isinstance(first_not_none, Series):
             # values are not series or array-like but scalars
             # self._selection_name not passed through to Series as the
             # result should not take the name of original selection
             # of columns
-            return self.obj._constructor_sliced(values, index=key_index)
+            if self.as_index:
+                return self.obj._constructor_sliced(values, index=key_index)
+            else:
+                result = DataFrame(values, index=key_index, columns=[self._selection])
+                self._insert_inaxis_grouper_inplace(result)
+                return result
```

Comment:
looks like this line isnt covered (the equivalent line isnt covered in master, so not necessarily a problem for this PR).  any chance it is unreachable and can be removed?

Reply:
Added test to cover it.